clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -990,6 +990,9 @@ def warn_main_redefined : Warning<"variable named 'main' with external linkage "
     "has undefined behavior">, InGroup<Main>;
 def ext_main_used : Extension<
     "referring to 'main' within an expression is a Clang extension">, InGroup<Main>;
+def err_invalid_linkage_specification : Extension<
+  "invalid linkage specification "
+  "'extern \"%select{C|C++}0\"'">;
```

Comment:
This doesn't say why it's invalid. `'main' cannot have linkage specification 'extern "C(++)"'`?

Reply:
@MitalAshok I've updated the diagnostic message